### PR TITLE
Issue/146 newline above blocks

### PR DESCRIPTION
**Pull request by theck13**
### Fix
Remove extra newline preceding block elements as described in https://github.com/wordpress-mobile/WordPress-Aztec-Android/issues/146.

### Test
1. Tap ***HTML*** format button.
2. Enter HTML shown in screenshots in this [comment](https://github.com/wordpress-mobile/WordPress-Aztec-Android/issues/146#issuecomment-267489480).
3. Tap ***HTML*** format button.

### CODE REVIEW

File: aztec/src/test/kotlin/org/wordpress/aztec/AztecParserTest.kt [hunk 1]
```diff
@@ -87,37 +91,68 @@ class AztecParserTest : AndroidTestCase() {
...
-                        HTML_NESTED_WITH_TEXT +
-                        HTML_NESTED_INTERLEAVING
+                HTML_HEADING_ALL +
+                HTML_BOLD +
+                HTML_ITALIC +
+                HTML_UNDERLINE +
+                HTML_STRIKETHROUGH +
+                HTML_LIST_ORDERED +
+                HTML_LIST_ORDERED_WITH_EMPTY_ITEM +
+//                HTML_LIST_ORDERED_WITH_QUOTE +
```
Comment (catehstn) [inline aztec/src/test/kotlin/org/wordpress/aztec/AztecParserTest.kt]: Why is this commented out? Should it be deleted?

File: aztec/src/test/kotlin/org/wordpress/aztec/AztecParserTest.kt [hunk 2]
```diff
@@ -87,37 +91,68 @@ class AztecParserTest : AndroidTestCase() {
...
+                HTML_LIST_UNORDERED +
+                HTML_LIST_UNORDERED_WITH_EMPTY_ITEM +
+//                HTML_LIST_UNORDERED_WITH_QUOTE +
```
Comment (catehstn) [inline aztec/src/test/kotlin/org/wordpress/aztec/AztecParserTest.kt]: and this

File: aztec/src/test/kotlin/org/wordpress/aztec/AztecParserTest.kt [hunk 3]
```diff
@@ -87,37 +91,68 @@ class AztecParserTest : AndroidTestCase() {
...
+
+        val parsed =
+                HTML_HEADING_ALL_PARSED +
+                HTML_BOLD +
+                HTML_ITALIC +
+                HTML_UNDERLINE +
+                HTML_STRIKETHROUGH +
+                HTML_LIST_ORDERED +
+                HTML_LIST_ORDERED_WITH_EMPTY_ITEM +
+//                HTML_LIST_ORDERED_WITH_QUOTE +
```
Comment (catehstn) [inline aztec/src/test/kotlin/org/wordpress/aztec/AztecParserTest.kt]: and here

File: aztec/src/test/kotlin/org/wordpress/aztec/AztecParserTest.kt [hunk 4]
```diff
@@ -87,37 +91,68 @@ class AztecParserTest : AndroidTestCase() {
...
+                HTML_LIST_UNORDERED +
+                HTML_LIST_UNORDERED_WITH_EMPTY_ITEM +
+//                HTML_LIST_UNORDERED_WITH_QUOTE +
```
Comment (catehstn) [inline aztec/src/test/kotlin/org/wordpress/aztec/AztecParserTest.kt]: ditto

File: aztec/src/test/kotlin/org/wordpress/aztec/AztecParserTest.kt [hunk 5]
```diff
@@ -87,37 +91,68 @@ class AztecParserTest : AndroidTestCase() {
...
+                HTML_NESTED_EMPTY_END +
+                HTML_NESTED_EMPTY_START +
+                HTML_NESTED_EMPTY +
+                HTML_NESTED_WITH_TEXT +
+                HTML_NESTED_INTERLEAVING
 
         val span = SpannableString(mParser.fromHtml(input, context))
         val output = mParser.toHtml(span)
-        Assert.assertEquals(input, output)
+        Assert.assertEquals(parsed, output)
```
Comment (catehstn) [inline aztec/src/test/kotlin/org/wordpress/aztec/AztecParserTest.kt]: I really like these tests! super straightforward :)

**Comment by theck13** [comment]
I added `HTML_LIST_ORDERED_WITH_QUOTE` and `HTML_LIST_UNORDERED_WITH_QUOTE` to the `parseHtmlToSpanToHtmlAll_isEqual` test to give it full coverate since they were missing.  That caused the test to fail.  Since the failure was not caused by these changes and to avoid holding up this pull request, I created a separate [issue](https://github.com/wordpress-mobile/WordPress-Aztec-Android/issues/160) to address them.  I left them commented as a reminder.  I guess adding `// TODO: ` would have been better.

File: aztec/src/test/kotlin/org/wordpress/aztec/AttributeTest.kt [hunk 6]
```diff
@@ -43,6 +43,7 @@ class AttributeTest {
         private val LIST_WITH_NON_EMPTY_ITEMS = "<ol><li a=\"B\">ab</li><li c=\"C\">c</li></ol>"
         private val LIST_WITH_EMPTY_ITEMS = "a<ul><li a=\"A\"></li><li></li><li a=\"1\">1</li><li></li><li b=\"B\"></li></ul>b"
         private val LIST_WITH_EMPTY_ITEMS_WITH_LINE_BREAK = "a<br><ul><li></li><li a=\"1\">1</li><li></li></ul><br>b"
+        private val LIST_WITH_EMPTY_ITEMS_WITH_LINE_BREAK_PARSED = "a<ul><li></li><li a=\"1\">1</li><li></li></ul><br>b"
```
Comment (0nko) [inline aztec/src/test/kotlin/org/wordpress/aztec/AttributeTest.kt]: Why are all these `<br>` being removed in the _PARSED strings?

Comment (theck13) [inline aztec/src/test/kotlin/org/wordpress/aztec/AttributeTest.kt]: The [`markBlockElementLineBreaks`](https://github.com/wordpress-mobile/WordPress-Aztec-Android/blob/develop/aztec/src/main/kotlin/org/wordpress/aztec/AztecParser.kt#L150) method removes the preceding `<br>` by translating it into a `BlockElementLinebreak` span.

Comment (0nko) [inline aztec/src/test/kotlin/org/wordpress/aztec/AttributeTest.kt]: But that's deleting a legitimate `<br>` above lists, isn't it? Say you have `a<br><br><ul><li>1</li></ul>` - when you switch between visual and HTML twice, all line breaks get removed.

**Pull request merged**